common/src/main/java/io/netty/util/NettyRuntime.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.util;
+
+import io.netty.util.internal.SystemPropertyUtil;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A utility class for wrapping calls to {@link Runtime}.
+ */
+public final class NettyRuntime {
+
+    private static final AtomicInteger availableProcessors = new AtomicInteger();
```

Comment:
Ok. Will try expand opinion.
Your solution looks like hack for solving a particular problem. It violates the [dependency inversion principle](https://en.wikipedia.org/wiki/Dependency_inversion_principle) ('D' from [SOLID](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design))) because enforce deepen in the implementation details. If someone wants to do a similar thing (configuring property 'availableProcessors' in runtime) and call setAvailableProcessors(), it can step on the rake and will be forced to classloading problem. This possibility forced you to create constraints (forbiddenapis).

Also if you need to do such a thing for the other Netty setting, you can't use reuse current code, and will be forced to create same hack.

Instead it, would be possible to modify SystemPropertyUtil for change/override/extend the property source (for example, for getting some properties from xml config or database).